vms/platformvm/state/state.go

```diff
@@ -2516,7 +2525,13 @@ func (s *state) writeChains() error {
 	return nil
 }
 
-func (s *state) writeMetadata() error {
+func (s *state) writeMetadata(height uint64) error {
+	if !s.upgrades.IsEtnaActivated(s.persistedTimestamp) && s.upgrades.IsEtnaActivated(s.timestamp) {
+		if err := database.PutUInt64(s.singletonDB, EtnaHeightKey, height); err != nil {
+			return fmt.Errorf("failed to write etna height: %w", err)
```

Comment:
nit: should we check if we have already written it? In case where we can call the commit without advancing the chain (like disconnect?).

Reply:
`persistedTimestamp` is updated to equal `timestamp` immediately after this, and both of them are monotonically increasing. So this condition can only happen once.